src/Umbraco.Web/Security/Providers/UmbracoMembershipProvider.cs

```diff
@@ -450,22 +456,15 @@ protected override string PerformResetPassword(string username, string answer, s
         /// </returns>
         public override bool UnlockUser(string username)
         {
-            var member = MemberService.GetByUsername(username);
-
-            if (member == null)
-            {
-                throw new ProviderException(string.Format("No member with the username '{0}' found", username));
-            }                
-
-            // Non need to update
-            if (member.IsLockedOut == false) return true;
-
-            member.IsLockedOut = false;
-            member.FailedPasswordAttempts = 0;
-
-            MemberService.Save(member);
+            var userManager = GetBackofficeUserManager();
+            return userManager != null && userManager.UnlockUser(username);
+        }
 
-            return true;
+        internal BackOfficeUserManager<BackOfficeIdentityUser> GetBackofficeUserManager()
+        {
+            return HttpContext.Current == null
```

Comment:
This means that this will only ever work in the context of a web request which is not ideal. 

In theory - these raising methods could be static since the events themselves are static